libretro-common/file/file_path.c

```diff
@@ -1149,6 +1149,44 @@ size_t fill_pathname_abbreviate_special(char *out_path,
    return strlcpy(out_path, in_path, size);
 }
 
+/**
+ * sanitize_path_part:
+ *
+ * @path_part               : directory or filename
+ *
+ * Takes single part of a path eg. single filename 
+ * or directory, and removes any special chars that are
+ * unavailable.
+ * 
+ * @returns new string that has been sanitized
+ **/
+const char *sanitize_path_part(const char *path_part)
+{
+   int i;
+   int j = 0;
+   int len = 0;
+   char *temp = NULL;
+   const char *special_chars = "<>:\"/\\|?*";
+
+   if (string_is_empty(path_part))
+      return NULL;
+
+   len = strlen(path_part);
```

Comment:
Can you pass the 'len' variable as a parameter to the function instead of doing an implicit strlen here?

This way, the caller can pass the size instead of it being explicitly calculated in this function.

Reply:
Resolved in newest commit.